gax-java/gax/src/main/java/com/google/api/gax/rpc/ClientContext.java

```diff
@@ -161,15 +161,16 @@ public static ClientContext create(StubSettings settings) throws IOException {
 
     Credentials credentials = settings.getCredentialsProvider().getCredentials();
     boolean usingGDCH = credentials instanceof GdchCredentials;
+    // Add two additional fields to the EndpointContext: 1. ServiceName and 2. usingGDCH field.
+    // The serviceName is overwritten in the child subclasses (GAPIC libraries will override the
+    // default StubSettings' serviceName). Client can determine if the GDC-H is being used via the
+    // Credentials. The Credentials object is resolved here and must be passed to the
+    // EndpointContext.
     EndpointContext endpointContext =
-        EndpointContext.newBuilder()
+        settings
+            .getEndpointContext()
+            .toBuilder()
             .setServiceName(settings.getServiceName())
```

Comment:
The alternative to this is to create a new private static method inside the {Client}StubSettings class and set it there.  Currently, we can't set it inside with the non-static `getServiceName()` method as initializing the defaults is inside a static method: https://github.com/googleapis/sdk-platform-java/blob/c1cd0d1deb74d7aa81f1ad7f6400d193f1464fac/showcase/gapic-showcase/src/main/java/com/google/showcase/v1beta1/stub/EchoStubSettings.java#L700-L710